c/wrapped_context.go

```diff
@@ -78,11 +88,11 @@ func (wctx *C.wrapped_context_t) NotifyError(result C.result_type_t, errorMessag
 	C.call_notify(wctx)
 }
 
-func (wctx *C.wrapped_context_t) UpdateMetadata(fileName string, length int64, downloadId int) {
+func (wctx *C.wrapped_context_t) UpdateMetadata(fileName string, length int64) {
+	wctx.Log("Updating metadata. Filename:%s, length:%d", fileName, length)
 	wctx.metadata.length = C.int64_t(length)
 	wctx.metadata.file_name = C.CString(fileName)
-	wctx.metadata.context = wctx.clientCtx
-	wctx.metadata.download_id = C.int32_t(downloadId)
+	wctx.metadata.context = wctx
```

Comment:
What is the idea behind saving context as part of the metadata?

Reply:
The idea was that once the metadata is received a Reject or Accept function can be derived from the metadata. However, that does make the API more confusing. I'm removing that field. Good catch!

Reply:
Similarly, `result_t` does have the same field with the same underlying concept, so I'll remove it from there too